smmregrid/gridtype.py

```diff
@@ -0,0 +1,214 @@
+# GridType class to gather all information about grids with shared dimensions
+import xarray as xr
+
+# default spatial dimensions and vertical coordinates
+DEFAULT_DIMS = {
+    'horizontal': ['i', 'j', 'x', 'y', 'lon', 'lat', 'longitude', 'latitude',
+                   'cell', 'cells', 'ncells', 'values', 'value', 'nod2', 'pix', 'elem',
+                   'nav_lon', 'nav_lat'],
+    'vertical': ['lev', 'nz1', 'nz', 'depth', 'depth_full', 'depth_half'],
+    'time': ['time']
+}
+
+
+class GridType:
```

Comment:
Here dimensions are clssified by name. An alternative way to check if a variable is horizontal, vertical etc would be to check its units (m, Pa, hPA, degrees etc)

Reply:
Maybe a later PR?